gateway/throttler/throttler.go

```diff
@@ -0,0 +1,135 @@
+package throttler
+
+//go:generate mockgen -destination=../../mocks/gateway/throttler.go -package=mocks_gateway github.com/rudderlabs/rudder-server/gateway/throttler GetThrottler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/internal/throttling"
+	"github.com/rudderlabs/rudder-server/services/stats"
+)
+
+const (
+	throttlingAlgoTypeGCRA = "gcra"
+)
+
+type Limiter interface {
+	// Allow returns true if the limit is not exceeded, false otherwise.
+	Allow(ctx context.Context, cost, rate, window int64, key string) (bool, func(context.Context) error, error)
+}
+
+type GetThrottler interface {
```

Comment:
Why is it called `GetThrottler` instead of `Throttler`?

Reply:
Very bad with naming 😕 changing it now